src/split/src/Split.tsx

```diff
@@ -72,8 +76,23 @@ export default defineComponent({
 
     const resizeTriggerElRef = ref<HTMLElement | null>(null)
     const isDraggingRef = ref(false)
-    const currentSize = ref(props.defaultSize)
-
+    const currentSize = ref(props.size || props.defaultSize)
+    if (props.size) {
+      watch(
+        () => props.size,
+        (newSize) => {
+          if (newSize <= props.max && newSize >= props.min) {
+            currentSize.value = newSize
+          }
+        }
+      )
+      watch(
+        () => currentSize.value,
+        (newSize) => {
+          emit('update:size', newSize)
+        }
+      )
```

Comment:
if props.size changes it will trigger currentSize and will trigger @update:size.

Reply:
i guess you can copy paste some code from checkbox for default and prop merging:
https://github.com/tusen-ai/naive-ui/blob/fce7b64ead4ba18d26b0115dc500303fdd70b3fc/src/checkbox/src/Checkbox.tsx#L153